### ISSUE TITLE
[HIGH] build_nstream_response returns fabricated logprobs from an English-word-length heuristic (estimate_logprob), not model logits

### ISSUE THREAD

**Issue by bong-water-water-bong** [1bit-MONSTER/1bit-MONSTER#81]
## Location
`src/server/rest_handler.cpp` — `RestHandler::build_nstream_response()` (line ~571) using `simple_tokenize()` (line ~411) and `estimate_logprob()` (line ~437)

## Description

The non-streaming OpenAI chat-completion path returns a `logprobs` object that is **not computed from the model's logits at all**. It is fabricated by a hand-rolled English-heuristic function:

```cpp
// src/server/rest_handler.cpp
static double estimate_logprob(const std::string& token) {
    static const std::unordered_set<std::string> common = {
        "the", "a", "an", "is", "are", ... ".", ",", "!", ...
    };
    ...
    if (common.count(clean)) return -0.10;      // function words → -0.10
    bool is_number = ...; if (is_number) return -0.50;
    if (clean.length() <= 2) return -0.30;
    if (clean.length() <= 4) return -0.50;
    if (clean.length() <= 6) return -0.80;
    if (clean.length() <= 8) return -1.50;
    return -3.00;                                // longer tokens → -3.0
}

json RestHandler::build_nstream_response(std::string response_text) {
    ...
    auto tokens = simple_tokenize(content);              // whitespace split, not real tokenization
    json logprobs_content = json::array();
    for (const auto& [tok, offset] : tokens) {
        double lp = estimate_logprob(tok);               // fabricated
        logprobs_content.push_back({
            {"token", tok},
            {"logprob", lp},
            {"bytes", std::vector<uint8_t>(tok.begin(), tok.end())},
            {"top_logprobs", nullptr}
        });
    }
    json logprobs_obj = { {"content", logprobs_content} };
    ...
}
```

`simple_tokenize` is just a whitespace/punctuation splitter — it returns nothing resembling the model's actual BPE token IDs, and `estimate_logprob` returns one of six hardcoded constants based purely on whether the lowercased token is in an English function-word set or how long it is. Neither has any contact with the model's `lm_head` logits.

The OpenAI spec requires `logprobs` (when `logprobs: true` is requested) to come from the model's actual output distribution. Note also that `build_nstream_response` is called **unconditionally** for every non-streaming OpenAI chat response — the `logprobs` object is always attached, even when the client did not request it.

## Impact

- Any client that requests `logprobs: true` (for calibration, perplexity eval, classifier-guided decoding, hallucination detection, etc.) receives **plausible-looking but entirely fabricated numbers**. A 6-char common word and a 6-char rare nonce both get `-0.80`; a 2-char token is always `-0.30`. This is silent — there's no warning that these are synthetic.
- Because `top_logprobs` is always `null` and the `token`/`bytes` fields are whitespace tokens (not model tokens), clients that try to correlate logprobs with sampled token IDs will also mismatch.
- The `token` strings in the logprobs object do not correspond to the model's tokenizer vocabulary, so any downstream lookup against the model vocab fails.

## Suggested fix direction

- If logprobs aren't actually implemented, do not emit a `logprobs` field at all (or emit `null`), and return an error / ignore when the client passes `logprobs: true`, rather than silently fabricating values.
- To implement for real: capture per-step argmax logits from `auto_chat_engine->generate(...)` (it currently emits only the argmax token text via the ostream), accumulate them, and index them against the model's tokenizer. This requires plumbing the logit vector out of the engine, which the current ostream interface doesn't expose.

## Related
Not covered by existing issues. The site numbers bug (#49) is about a different generator/JS field mismatch; this is about the runtime API lying about logprobs.

**Comment by bong-water-water-bong** [comment]
Partially fixed in `9791b8d` (pushed to main) — the silent lie is removed; real logprobs are still not implemented.

`build_nstream_response` now emits `"logprobs": null` (the OpenAI default) instead of the fabricated `estimate_logprob` object, and the dead `simple_tokenize` / `estimate_logprob` helpers + the `unordered_set` include were removed:

```diff
-    json logprobs_obj = { {"content", logprobs_content} };
-    return json::array({ { ..., {"logprobs", logprobs_obj}, ... } });
+    return json::array({ { ..., {"logprobs", nullptr}, ... } });
```

This stops the API from returning fabricated numbers. **What's still open:** honoring `logprobs: true` for real requires plumbing per-step argmax logits out of `auto_chat_engine->generate(...)`, which the current ostream interface doesn't expose — that's new implementation work, not a one-line fix, so the issue stays open until either real logprob capture lands or the endpoint explicitly rejects `logprobs: true` with a clear error.
